packages/next/src/generators/init/utils/infrastructure/terraform.ts

```diff
@@ -0,0 +1,223 @@
+                },
+            ],
+            cwd: `${project.root}/build/terraform`,
+        },
+    };
+
+    updateProjectConfiguration(tree, project.name, update);
+
+    updateJson(tree, 'nx.json', nxJson => {
+        const updateNxJson = { ...nxJson };
+
+        updateNxJson.tasksRunnerOptions.default.options.cacheableOperations = [
+            ...new Set([
+                ...nxJson.tasksRunnerOptions.default.options
+                    .cacheableOperations,
+                'terraform-fmt',
+                'terraform-init',
+                'terraform-validate',
+                'terraform-plan',
+                'terraform-apply',
+            ]),
+        ];
+
+        if (!nxJson.targetDefaults['terraform-fmt']) {
+            updateNxJson.targetDefaults['terraform-fmt'] = {
+                inputs: ['terraform'],
+            };
+        }
+
+        if (!nxJson.targetDefaults['terraform-init']) {
+            updateNxJson.targetDefaults['terraform-init'] = {
+                inputs: ['terraform'],
+                outputs: [
+                    '{projectRoot}/build/terraform/.terraform',
+                    '{projectRoot}/build/terraform/.terraform.lock.hcl',
+                ],
+            };
+        }
+
+        if (!nxJson.targetDefaults['terraform-validate']) {
+            updateNxJson.targetDefaults['terraform-validate'] = {
+                dependsOn: ['terraform-init'],
+                inputs: ['terraform'],
+            };
+        }
+
+        if (!nxJson.targetDefaults['terraform-plan']) {
+            updateNxJson.targetDefaults['terraform-plan'] = {
+                dependsOn: ['terraform-init'],
+                inputs: ['terraform'],
+                outputs: ['{projectRoot}/build/terraform/*.tfplan'],
+            };
+        }
+
+        if (!nxJson.targetDefaults['terraform-apply']) {
+            updateNxJson.targetDefaults['terraform-apply'] = {
+                dependsOn: ['terraform-plan'],
+                inputs: ['terraform'],
+            };
+        }
+
+        if (!nxJson.namedInputs.terraform) {
+            updateNxJson.namedInputs.terraform = [
+                '{projectRoot}/build/terraform/*.tf',
+                '{projectRoot}/build/terraform/*.tfvars',
+                '{projectRoot}/build/terraform/.terraform.lock.hcl',
+            ];
+        }
+
+        return updateNxJson;
+    });
+
+    return () => {};
+}
+
+// namedInputs: {
+//     default: [
+//         ...new Set([
+//             ...nxJson.namedInputs.default,
+//             '!{projectRoot}/Dockerfile',
+//             '!{projectRoot}/Chart.yaml',
+//             '!{projectRoot}/values.yaml',
+//             '!{projectRoot}/charts/**/*',
+//             '!{projectRoot}/templates/**/*',
+//         ]),
+//     ],
+//     terraform: [
+//         '{projectRoot}/build/terraform/*.tf',
+//         '{projectRoot}/build/terraform/*.tfvars',
+//         '{projectRoot}/build/terraform/.terraform.lock.hcl',
+//     ],
+// },
```

Comment:
Is this on purpose? If so can you add a comment description explaining why

Reply:
Ta, forgot to cleanup :D